lib/matplotlib/ticker.py

```diff
@@ -2342,22 +2351,25 @@ def tick_values(self, vmin, vmax):
         # Get decades between major ticks.
         stride = (max(math.ceil(numdec / (numticks - 1)), 1)
                   if mpl.rcParams['_internal.classic_mode'] else
-                  (numdec + 1) // numticks + 1)
-
-        # if we have decided that the stride is as big or bigger than
-        # the range, clip the stride back to the available range - 1
-        # with a floor of 1.  This prevents getting axis with only 1 tick
-        # visible.
-        if stride >= numdec:
-            stride = max(1, numdec - 1)
+                  numdec // numticks)
+
+        if stride > numdec:
+            # If we have decided that the stride is bigger than the range, clip
+            # the stride back to the available range - 1 with a floor of 1.
+            # This prevents getting axis with only 1 tick visible.
+            stride = max(1, numdec)
```

Comment:
code cov thinks this needs a test, and since its new, it is probably correct....

Reply:
Thanks for spotting - it turns out this bit of code is redundant now and never hit, so I've removed it.